pkgs/applications/virtualization/qemu/default.nix

```diff
@@ -143,6 +148,10 @@ stdenv.mkDerivation (finalAttrs: {
       sha256 = "sha256-oC+bRjEHixv1QEFO9XAm4HHOwoiT+NkhknKGPydnZ5E=";
       revert = true;
     })
+
+    # qemu defined a crc32c function which clashes with one in libblkid when
+    # building statically
+    ./static_build_crc32c_duplicate_definition.patch
```

Comment:
Do we really need whatever is enabling this? I'd rather not add another downstream patch unless there's a very clear need.

Reply:
Alternatively: can it be upstreamed?

Reply:
At the very least, this patch should be a `lib.lists.optional` conditional that checks for the static build. There's no need to enable this on every QEMU build.

Reply:
> Do we really need whatever is enabling this? I'd rather not add another downstream patch unless there's a very clear need.

what do you mean ?

> At the very least, this patch should be a lib.lists.optional conditional that checks for the static build. There's no need to enable this on every QEMU build.

to the contrary patches should be eagerly applied otherwise they bitrot.

Reply:
> > Do we really need whatever is enabling this? I'd rather not add another downstream patch unless there's a very clear need.
> 
> what do you mean ?

I mean, given we can build a subset of static qemu without this (this is now used for binfmt), then whatever feature requires this must be optional.  So is it one that we need, or can we just disable it for static builds and avoid the patch?